skeleton/src/main/java/com/dropbox/kaiken/skeleton/usermanagement/ActiveUserDataSource.kt

```diff
@@ -1,64 +1,50 @@
-package com.dropbox.kaiken.skeleton.usermanagement.auth
+package com.dropbox.kaiken.skeleton.usermanagement
 
 import android.app.Application
-import android.content.Context.MODE_PRIVATE
+import android.content.Context
 import android.content.SharedPreferences
 import com.dropbox.kaiken.skeleton.scoping.AppScope
-import com.dropbox.kaiken.skeleton.scoping.SingleIn
 import com.squareup.anvil.annotations.ContributesBinding
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.onStart
 import javax.inject.Inject
 
-interface ActiveUserManager {
-
-    /**
-     * Clears the active users
-     *
-     * @return true if the active user was set and cleared
-     */
-    suspend fun clearActiveUser(): Boolean
-
-    /**
-     * Sets the active users and persists to disk
-     *
-     */
+interface ActiveUserDataSource {
     suspend fun setActiveUser(userId: String)
-
-    /**
-     * Get the active user from disk
-     *
-     * @return the active user ID, or null if none has been set
-     */
     suspend fun getActiveUserId(): String?
+    suspend fun clearActiveUser(): Boolean
+    fun getActiveUser(): Flow<String?>
 }
+
 @ContributesBinding(AppScope::class)
-@SingleIn(AppScope::class)
-class RealActiveUserManager @Inject constructor(
-    private val context: Application
-) : ActiveUserManager {
+class RealActiveUserDataSource @Inject constructor(application: Application) : ActiveUserDataSource {
```

Comment:
should the real impl be in sample only?

Reply:
Yup - I think this is moved entirely out of skeleton / kaiken.  Active user is entirely an app related feature.